File: Leetcode/Easy/1768. Merge Strings Alternately/solution.ts
function mergeAlternately(word1: string, word2: string): string {
    
    let i = 0, ans = [];
    while(i < word1.length || i < word2.length) {
        if (i < word1.length) {
            ans.push(word1.split("")[i])
        }
        if (i < word2.length) {
            ans.push(word2.split("")[i])
        }

        i++;
    } 

    return ans.join("");
};

File: Leetcode/Easy/1768. Merge Strings Alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        # ans stack
        # loop while stack len less than word1 len + word2 len
        # push to stack first word1 then word2

        stack = []
        i = 0
        while len(stack) < len(word1) + len(word2):

            if i < len(word1):
                stack.append(word1[i])

            if i < len(word2):
                stack.append(word2[i])

            i+=1

        s = ""

        return s.join(stack)


